src/components/SettingsPage/SettingsSyncCard.tsx

```diff
@@ -83,20 +84,29 @@ export const SettingsSyncCard = () => {
   >(null);
 
   const [lastSyncedTime, setLastSyncedTime] = useState<string | null>(null);
+  const [forbiddenError, setForbiddenError] = useState<boolean>(false);
 
   const submitEntitySync = useCallback(async () => {
     const { items: entities } = await catalogApi.getEntities();
     const shouldGzipBody =
       config.getOptionalBoolean('cortex.syncWithGzip') ?? false;
     const customMappings = await extensionApi.getCustomMappings?.();
     const groupOverrides = await extensionApi.getTeamOverrides?.(entities);
-    const progress = await cortexApi.submitEntitySync(
-      entities,
-      shouldGzipBody,
-      customMappings,
-      groupOverrides,
-    );
-    setSyncTaskProgressPercentage(progress.percentage);
+    try {
```

Comment:
Is there a way to know whether the user has privileges beforehand and thus prevent them from clicking the button in the first place?

Reply:
yes that's a good idea - just pushed a change for this to fetch a user's permissions upfront and display the settings page accordingly see `CortexPage.tsx`